src/app/codegen-data-model-provider/CodegenDataModelProvider.cpp

```diff
@@ -536,6 +585,47 @@ std::optional<DataModel::ClusterInfo> CodegenDataModelProvider::GetClusterInfo(c
     return std::make_optional(std::get<DataModel::ClusterInfo>(info));
 }
 
+ClusterId CodegenDataModelProvider::FirstClientCluster(EndpointId endpointId)
+{
+    const EmberAfEndpointType * endpoint = emberAfFindEndpointType(endpointId);
+    VerifyOrReturnValue(endpoint != nullptr, kInvalidClusterId);
+    VerifyOrReturnValue(endpoint->clusterCount > 0, kInvalidClusterId);
+    VerifyOrReturnValue(endpoint->cluster != nullptr, kInvalidClusterId);
+
+    return FirstClientClusterId(endpoint, 0, mClientClusterIterationHint);
+}
+
+ClusterId CodegenDataModelProvider::NextClientCluster(const ConcreteClusterPath & before)
+{
+    const EmberAfEndpointType * endpoint = emberAfFindEndpointType(before.mEndpointId);
+    VerifyOrReturnValue(endpoint != nullptr, kInvalidClusterId);
+    VerifyOrReturnValue(endpoint->clusterCount > 0, kInvalidClusterId);
+    VerifyOrReturnValue(endpoint->cluster != nullptr, kInvalidClusterId);
+
+    if (mServerClusterIterationHint < endpoint->clusterCount)
```

Comment:
This line makes no sense.

Reply:
Oh, this should be `mClientClusterIterationHint`, changed it.